modules/atmosphere/rendering/atmospheredeferredcaster.cpp

```diff
@@ -0,0 +1,1595 @@
+    , _deltaSRayleighTableTexture(0)
+    , _deltaSMieTableTexture(0)
+    , _deltaJTableTexture(0)
+    , _atmosphereTexture(0)
+    , _atmosphereDepthTexture(0)
+    , _atmosphereFBO(0)
+    , _atmosphereRenderVAO(0)
+    , _atmosphereRenderVBO(0)
+    , _transmittance_table_width(256)
+    , _transmittance_table_height(64)
+    , _irradiance_table_width(64)
+    , _irradiance_table_height(16)
+    , _delta_e_table_width(64)
+    , _delta_e_table_height(16)
+    , _r_samples(32)
+    , _mu_samples(128)
+    , _mu_s_samples(32)
+    , _nu_samples(8)
+    , _atmosphereCalculated(false)
+    , _atmosphereEnabled(false)
+    , _ozoneEnabled(false)
+    , _sunFollowingCameraEnabled(false)
+    , _atmosphereRadius(0.f)
+    , _atmospherePlanetRadius(0.f)
+    , _planetAverageGroundReflectance(0.f)
+    , _rayleighHeightScale(0.f)
+    , _ozoneHeightScale(0.f)
+    , _mieHeightScale(0.f)
+    , _miePhaseConstant(0.f)    
+    , _rayleighScatteringCoeff(glm::vec3(0.f))
+    , _ozoneExtinctionCoeff(glm::vec3(0.f))
+    , _mieScatteringCoeff(glm::vec3(0.f))
+    , _mieExtinctionCoeff(glm::vec3(0.f))
+    , _ellipsoidRadii(glm::dvec3(0.0))
+    , _sunRadianceIntensity(50.0f)
+    , _exposureConstant(0.4f)
+    , _exposureBackgroundConstant(1.8f)
+    , _gammaConstant(1.8f)
+    , _renderableClass(NoRenderableClass)
+    , _calculationTextureScale(1.0)
+    , _saveCalculationTextures(false)
+
+
+{}
+
+AtmosphereDeferredcaster::~AtmosphereDeferredcaster() {}
+
+void AtmosphereDeferredcaster::initialize()
+{
+    if (!_atmosphereCalculated) {
+        preCalculateAtmosphereParam();
+    }
+}
+
+void AtmosphereDeferredcaster::deinitialize()
+{
+    RenderEngine& renderEngine = OsEng.renderEngine();
+
+    if (_transmittanceProgramObject) {
+        _transmittanceProgramObject = nullptr;
+    }
+
+    if (_irradianceProgramObject) {
+        _irradianceProgramObject = nullptr;
+    }
+
+    if (_irradianceSupTermsProgramObject) {
+        _irradianceSupTermsProgramObject = nullptr;
+    }
+
+    if (_inScatteringProgramObject) {
+        _inScatteringProgramObject = nullptr;
+    }
+
+    if (_inScatteringSupTermsProgramObject) {
+        _inScatteringSupTermsProgramObject = nullptr;
+    }
+
+    if (_deltaEProgramObject) {
+        _deltaEProgramObject = nullptr;
+    }
+
+    if (_deltaSProgramObject) {
+        _deltaSProgramObject = nullptr;
+    }
+
+    if (_deltaSSupTermsProgramObject) {
+        _deltaSSupTermsProgramObject = nullptr;
+    }
+
+    if (_deltaJProgramObject) {
+        _deltaJProgramObject = nullptr;
+    }
+
+    glDeleteTextures(1, &_transmittanceTableTexture);
+    glDeleteTextures(1, &_irradianceTableTexture);
+    glDeleteTextures(1, &_inScatteringTableTexture);
+    glDeleteTextures(1, &_deltaETableTexture);
+    glDeleteTextures(1, &_deltaSRayleighTableTexture);
+    glDeleteTextures(1, &_deltaSMieTableTexture);
+    glDeleteTextures(1, &_deltaJTableTexture);
+    glDeleteTextures(1, &_atmosphereTexture);
+
```

Comment:
No. These are the pre-calculated tables used in atmosphere rendering. I am being thorough here.